src/System.Private.Reflection.Core/src/System/Reflection/Runtime/PropertyInfos/EcmaFormat/EcmaFormatRuntimePropertyInfo.cs

```diff
@@ -0,0 +1,188 @@
+        //           The definingType and contextType are both Foo<>
+        //
+        //       typeof(Foo<int,String>).GetTypeInfo().DeclaredMembers
+        //
+        //          The definingType is "Foo<,>"
+        //          The contextType is "Foo<int,String>"
+        //
+        //  We don't report any DeclaredMembers for arrays or generic parameters so those don't apply.
+        //
+        private EcmaFormatRuntimePropertyInfo(PropertyDefinitionHandle propertyHandle, EcmaFormatRuntimeNamedTypeInfo definingTypeInfo, RuntimeTypeInfo contextTypeInfo, RuntimeTypeInfo reflectedType) :
+            base(contextTypeInfo, reflectedType)
+        {
+            _propertyHandle = propertyHandle;
+            _definingTypeInfo = definingTypeInfo;
+            _reader = definingTypeInfo.Reader;
+            _property = _reader.GetPropertyDefinition(propertyHandle);
+        }
+
+        public sealed override PropertyAttributes Attributes
+        {
+            get
+            {
+                return _property.Attributes;
+            }
+        }
+
+        public sealed override IEnumerable<CustomAttributeData> CustomAttributes
+        {
+            get
+            {
+#if ENABLE_REFLECTION_TRACE
+                if (ReflectionTrace.Enabled)
+                    ReflectionTrace.PropertyInfo_CustomAttributes(this);
+#endif
+
+                foreach (CustomAttributeData cad in RuntimeCustomAttributeData.GetCustomAttributes(_reader, _property.GetCustomAttributes()))
+                    yield return cad;
+            }
+        }
+
+        public sealed override bool Equals(Object obj)
+        {
+            EcmaFormatRuntimePropertyInfo other = obj as EcmaFormatRuntimePropertyInfo;
+            if (other == null)
+                return false;
+            if (!(_reader == other._reader))
+                return false;
+            if (!(_propertyHandle.Equals(other._propertyHandle)))
+                return false;
+            if (!(ContextTypeInfo.Equals(other.ContextTypeInfo)))
+                return false;
+            if (!(_reflectedType.Equals(other._reflectedType)))
+                return false;
+            return true;
+        }
+
+        public sealed override int GetHashCode()
+        {
+            return _propertyHandle.GetHashCode();
+        }
+
+        public sealed override Object GetConstantValue()
+        {
+#if ENABLE_REFLECTION_TRACE
+            if (ReflectionTrace.Enabled)
+                ReflectionTrace.PropertyInfo_GetConstantValue(this);
+#endif
+            Object defaultValue;
+            if (!DefaultValueProcessing.GetDefaultValueIfAny(_reader, ref _property, this, out defaultValue))
+            {
+                throw new InvalidOperationException();
+            }
+            return defaultValue;
+        }
+
+        public sealed override int MetadataToken
+        {
+            get
+            {
+                throw new InvalidOperationException(SR.NoMetadataTokenAvailable);
```

Comment:
Uh, that does seem silly, doesn't it. I've updated the various apis to make MetadataToken handled correctly for methods, fields, events, properties, types, and constructors